oauthproxy.go

```diff
@@ -761,48 +746,48 @@ func (p *OAuthProxy) OAuthCallback(rw http.ResponseWriter, req *http.Request) {
 	err := req.ParseForm()
 	if err != nil {
 		logger.Errorf("Error while parsing OAuth2 callback: %v", err)
-		p.ErrorPage(rw, req, http.StatusInternalServerError, "Internal Server Error", err.Error())
+		p.ErrorPage(rw, req, http.StatusInternalServerError, err.Error())
 		return
 	}
 	errorString := req.Form.Get("error")
 	if errorString != "" {
 		logger.Errorf("Error while parsing OAuth2 callback: %s", errorString)
-		p.ErrorPage(rw, req, http.StatusForbidden, "Permission Denied", errorString)
+		p.ErrorPage(rw, req, http.StatusForbidden, errorString, "Login Failed: The upstream identity provider returned an error.")
 		return
 	}
 
 	session, err := p.redeemCode(req)
 	if err != nil {
 		logger.Errorf("Error redeeming code during OAuth2 callback: %v", err)
-		p.ErrorPage(rw, req, http.StatusInternalServerError, "Internal Server Error", "Internal Error")
+		p.ErrorPage(rw, req, http.StatusInternalServerError, err.Error(), "Login Failed: There was a problem with the code returned by the upstream identity provider.")
```

Comment:
I think this message potentially gives too much details that a request is failing during the `token_endpoint` request with an invalid `authorization_code`

I think this & the CSRF check failing on a bad state should give the same generic error message to not tip an attacker off if they are attacking in some way as to which layer their attack failed.

Reply:
ToDo: Move to internal server error